### DISCUSSION TITLE
\parencite 的定义

### DISCUSSION THREAD

**Discussion by SwitWu** [stone-zeng/fduthesis#335] [General]
https://github.com/stone-zeng/fduthesis/blob/d84e0dd7df657d47943e98c98ff843cc1336bc5f/source/fduthesis.dtx#L5718-L5719

这里的定义略显复杂，请问这样做的目的是什么？`cite` 后面的空格 `~` 是为了干什么？

**Comment by stone-zeng** [comment]
`\cite` 本身也是用 `\DeclareRobustCommand` 定义的命令，这会把 `\cite` 定义为 `\protect\cite<space>`，而真实的命令定义则会存到 `\cite<space>` 里面。

所以这里先通过 `\cs:w cite ~ \cs_end:` 拿到 `\cite<space>`（注意 LaTeX3 语法里面 `~` 就是空格），然后再把它展开一次，实际上总的效果相当于

```tex
\DeclareRobustCommand\parencite
    {\begingroup\let\NAT@ctype\z@\NAT@partrue\NAT@swatrue
      \@ifstar{\NAT@fulltrue\NAT@cites}{\NAT@fullfalse\NAT@cites}}
```

也就是直接拷贝一下（`natbib` 中）原来 `cite` 的定义。
